lib/dock.rb

```diff
@@ -0,0 +1,44 @@
+class Dock
+
+  attr_reader :name,
+              :max_rental_time
+
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @rented_boats = []
+    @total_revenue = 0
+  end
+
+  def rent(boat, renter)
+    boat.rented_status = true
+    @rented_boats << boat
+  end
+
+  def return(boat)
+    boat.rented_status = false
+    # @rented_boats.delete(boat)
+  end
+
+  def log_hour
+    @rented_boats.each do |boat|
+      if boat.rented_status == true && boat.hours_rented < 3
+        # get feedback on the above boolean, what if both are false
```

Comment:
If both are false then this will not run. && means "both are true", not "they are the same"